### ISSUE TITLE
"Unable to find source for object c8f1076693c0......" when git push

### ISSUE THREAD

**Issue by KevinChann** [git-lfs/git-lfs#3912]
I got a problem when I push my mycode using `git push <git_path> <branch_name>`. But when I use `git push origin <branch_name>`, I push successfully.
For Example,
```
git clone git@github.com:git-lfs/git-lfs.git master
git lfs pull git@github.com:git-lfs/git-lfs.git master --include="*" --exclude=""
```

And,
It's ok when my command is `git push origin master`

It's **NOT** ok when my command is `git push git@github.com:git-lfs/git-lfs.git master` . 
The error message is **"Unable to find source for object c8f1076693c0...... (try running git lfs fetch --all): open /<my_path>/<a_deleted_bianry_file>: no such file or directory"**

**<a_deleted_bianry_file>** is a old file in repository that I already **DELETED**.

 It seems that the index  got some problems because when i execute the command `git filter-branch --force --index-filter 'git rm --cached --ignore-unmatch <a_deleted_bianry_file>' --prune-empty --tag-name-filter cat -- --all`, 
then I can `git push <git_path> <branch_name>` successfully.

This is my issue.  I will be looking forward to your reply.

**Comment by bk2204** [comment]
Hey,

Can you provide the output of `git lfs env` so we can see the versions of Git and Git LFS that you're using?  We've fixed some issues in recent versions that meant that upload attempts would need to have files that strictly weren't required for upload.  In most cases, we don't need those objects anymore in recent versions.

If you're pushing to a _different_ repository than the one you got your objects from, then you'll need to upload those objects to the new repository, even if they've been deleted, which you can do by running `git lfs fetch --all origin` and then pushing.  They're part of your history, and someone could clone the second repository and need to check out an old version, so we need to upload them.

**Comment by KevinChann** [comment]
`git lfs env`  version and some messages:
```
git-lfs/2.7.1 (GitHub; linux amd64; go 1.12; git 6b7fb6e3)
git version 1.8.3.1
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

It's the same repository with origin. And I try `git lfs fetch --all `, it just told me the branch is up-to-date or something like that.

**Comment by bk2204** [comment]
I think you're getting the up-to-date message by running `git fetch --all`, which is a little different from `git lfs fetch --all`.  The latter doesn't produce that message.

Regardless, I believe this issue has been fixed in 2.8.0 with #3634.  Can you try a newer version to see if that fixes your problem?

**Comment by bk2204** [comment]
Hey,

Since there hasn't been any response in over a month, I'm going to close this.  Please feel free to reopen if you can provide additional information.
